akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -726,3 +729,139 @@ private[akka] object Log {
   private final val OffInt = LogLevels.Off.asInt
   private final val DefaultLogLevels = LogLevels(onElement = Logging.DebugLevel, onFinish = Logging.DebugLevel, onFailure = Logging.ErrorLevel)
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] object TimerKeys {
+  case object TakeWithinTimerKey
+  case object DropWithinTimerKey
+  case object GroupedWithinTimerKey
+}
+
+/**
+ * INTERNAL API
+ */
+private[stream] final case class GroupedWithin[T](n: Int, d: FiniteDuration) extends TimerStage[T, immutable.Seq[T]] {
+  import TimerKeys._
+  private var buf: Vector[T] = Vector.empty
+
+  override def preStart(ctx: AsyncContext[immutable.Seq[T], Scheduled]) = {
+    super.preStart(ctx)
+    schedulePeriodically(GroupedWithinTimerKey, d)
```

Comment:
I think I'd prefer extending `AsyncContext` to be able to schedule in a normal `AsyncStage`, one less concept to pick from when implementing stuff, in order to avoid: *"hmm... timer is async stuff, I know async stage already. I guess it can schedule. Oh it can't... oh there's TimerStage."*